### PR TITLE
FE-865 - Component wrapper for Matchbox Banner

### PR DESCRIPTION
**Pull request by shrirupa**
FE-865 - Component wrapper for Matchbox Banner

### What Changed
 - Banner has a single, globally accessible point of entry within the app
 - Banner is only imported from Matchbox in one place - replaced all instances of the Banner directly imported from matchbox with this new global component
- New props available to the Hibana version of the component are exposed - and subsequently ignored by the original component (https://jira.int.messagesystems.com/browse/FE-812)
- Ensure no unusual positioning UI defects are introduced as a result of this change - both for when using Hibana and when using the default theme

### How To Test
 - enable hibana ui flag 
`{
	"options": {
		"ui":{
"hibana":{
	"hasThemeControls": true
}
		}
	}
 }`
- Check all instances of Banner for ui defects with and without hibana enabled

### CODE REVIEW

File: src/pages/trackingDomains/components/UnverifiedBanner.js [hunk 1]
```diff
@@ -9,14 +10,20 @@ const UnverifiedBanner = ({ unverifiedDomains, cname }) => {
...
+      ? `You have ${count} unverified tracking domains`
+      : 'You have an unverified tracking domain';
 
   return (
-    <Banner
-      status='warning'
-      title={title}>
-      <p>To verify a tracking domain, edit its DNS settings to <strong>add a CNAME record</strong> with the value of <strong>{cname}</strong>.</p>
-      <Button outline external to={LINKS.DOMAIN_VERIFICATION}>Learn more</Button>
+    <Banner status="warning" title={title} my={10}>
```
Comment (shrirupa) [inline src/pages/trackingDomains/components/UnverifiedBanner.js]: only this line changed.

File: src/pages/trackingDomains/CreatePage.js [hunk 2]
```diff
@@ -21,20 +21,32 @@ export class CreatePage extends Component {
...
+            content: 'Back to Tracking Domains',
+            to: '/account/tracking-domains',
+            Component: Link,
+          }}
         />
-        <Banner
-          status="info"
-          title="Verification required">
-          <p>Tracking domains need to be verified via DNS. You'll need to <strong>add a CNAME record</strong> with the value of <strong>{cname}</strong> to this domain's DNS settings before it can be used or set as the default. <UnstyledLink to={LINKS.DOMAIN_VERIFICATION} external>Learn more about editing your DNS settings.</UnstyledLink></p>
+        <Banner status="info" title="Verification required" my={10}>
```
Comment (shrirupa) [inline src/pages/trackingDomains/CreatePage.js]: this line changed

File: src/pages/sendingDomains/components/UnverifiedWarningBanner.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 
   return (
     <Banner
-      status='warning'
-      title='Unverified sending domains will be removed two weeks after creation.'
-      action={action} >
+      status="warning"
+      title="Unverified sending domains will be removed two weeks after creation."
+      action={action}
+      my={10}
```
Comment (shrirupa) [inline src/pages/sendingDomains/components/UnverifiedWarningBanner.js]: this line changed

File: src/pages/sendingDomains/components/EditBounce.js [hunk 4]
```diff
@@ -41,8 +42,9 @@ export class EditBounce extends Component {
     }
 
     return (
-      <Banner status="warning">
-        We strongly recommend using a subdomain such as <strong>bounces.{id}</strong> for bounce domains. <Link to={'/account/sending-domains/create'}>Create a new domain now.</Link>
+      <Banner status="warning" my={10}>
```
Comment (shrirupa) [inline src/pages/sendingDomains/components/EditBounce.js]: this line changed

File: src/pages/sendingDomains/components/DomainStatus.js [hunk 5]
```diff
@@ -14,25 +14,20 @@ export const DomainStatus = ({ domain, hasAutoVerifyEnabled, onShareDomainChange
...
       <SendingDomainSection.Left>
         <VerificationHelp status={status} />
       </SendingDomainSection.Left>
       <SendingDomainSection.Right>
         {hasAutoVerifyEnabled && status === 'verified' && (
-          <Banner
-            title="Auto-Verify is Enabled"
-            status="info"
-          >
+          <Banner title="Auto-Verify is Enabled" status="info" my={10}>
```
Comment (shrirupa) [inline src/pages/sendingDomains/components/DomainStatus.js]: this line changed

File: src/pages/recipientLists/components/RecipientListForm.js [hunk 6]
```diff
@@ -48,7 +49,7 @@ export class RecipientListForm extends Component {
   renderCsvErrors() {
     const { error } = this.props;
     return (
-      <Banner status="danger" title="CSV Format Errors">
+      <Banner status="danger" title="CSV Format Errors" my={10}>
```
Comment (shrirupa) [inline src/pages/recipientLists/components/RecipientListForm.js]: this line changed

File: src/pages/profile/components/BackupCodesModal.js [hunk 7]
```diff
@@ -27,69 +28,80 @@ export default class BackupCodesModal extends Component {
...
+        <Panel title="Generate Two-factor Backup Codes" accent>
+          <form onSubmit={e => e.preventDefault()}>
             <Panel.Section>
-              { !generatedCodes && hasCodes && <Banner status="warning" >
-                Clicking Generate will overwrite your existing {activeCount} active backup codes.
-              </Banner> }
-              <p>Keep these single-use backup codes somewhere safe but accessible.
-                They can be used if your authentication app is unavailable (<span role="img" aria-label="phone in toilet emojis">📱  ➡︎🚽</span> , etc).
+              {!generatedCodes && hasCodes && (
+                <Banner status="warning" my={10}>
```
Comment (shrirupa) [inline src/pages/profile/components/BackupCodesModal.js]: this line changed

File: src/pages/ipPools/ListPage.js [hunk 8]
```diff
@@ -88,7 +89,7 @@ export class IpPoolsList extends Component {
 }
 
 export const IPWarmupReminderBanner = () => (
-  <Banner status="warning" title={'New dedicated IP addresses need to be warmed up'}>
+  <Banner status="warning" title={'New dedicated IP addresses need to be warmed up'} my={10}>
```
Comment (shrirupa) [inline src/pages/ipPools/ListPage.js]: this line changed

File: src/pages/blacklist/components/CongratsBanner.js [hunk 9]
```diff
@@ -8,7 +8,7 @@ const CongratsBanner = ({
 }) => {
   return (
     <div data-id="congrats-banner">
-      <Banner title={title} status="success" onDismiss={onDismiss}>
+      <Banner title={title} status="success" onDismiss={onDismiss} my={10}>
```
Comment (shrirupa) [inline src/pages/blacklist/components/CongratsBanner.js]: this line changed

File: src/pages/billing/components/SuspendedForBilling.js [hunk 10]
```diff
@@ -8,7 +8,11 @@ export default function SuspendedForBilling({ account }) {
   const email = billing.email ? ` (${billing.email})` : '';
   return (
     <Fragment>
-      <Banner status="danger" title="Your account has been suspended due to a billing problem">
+      <Banner
+        status="danger"
+        title="Your account has been suspended due to a billing problem"
+        my={10}
```
Comment (shrirupa) [inline src/pages/billing/components/SuspendedForBilling.js]: this line changed

File: src/pages/billing/components/Banners.js [hunk 11]
```diff
@@ -118,7 +120,7 @@ export const FreePlanWarningBanner = ({
   const daysLeft = Math.floor(ageRangeEnd - accountAgeInDays) + 1;
 
   return (
-    <Banner status="warning" title="Free Plan Downgrade">
+    <Banner status="warning" title="Free Plan Downgrade" my={10}>
```
Comment (shrirupa) [inline src/pages/billing/components/Banners.js]: this line changed

File: src/pages/billing/components/Banners.js [hunk 12]
```diff
@@ -33,7 +33,7 @@ export const PendingPlanBanner = ({ account, subscription }) => {
   }
 
   return (
-    <Banner status="info" title="Pending Plan Change">
+    <Banner status="info" title="Pending Plan Change" my={10}>
```
Comment (shrirupa) [inline src/pages/billing/components/Banners.js]: this line changed

File: src/pages/billing/components/Banners.js [hunk 13]
```diff
@@ -23,7 +23,7 @@ export const PendingPlanBanner = ({ account, subscription }) => {
 
   if (pendingDowngrades.length > 0) {
     return (
-      <Banner status="info" title="Pending Plan Change">
+      <Banner status="info" title="Pending Plan Change" my={10}>
```
Comment (shrirupa) [inline src/pages/billing/components/Banners.js]: this line changed

File: src/pages/api-keys/DetailsPage.js [hunk 14]
```diff
@@ -31,22 +32,19 @@ export class ApiKeysDetailsPage extends Component {
...
   renderNotFound() {
-    return (<Banner
-      status='warning'
-      title='Not found'
-
-    >
-      <p>API Key was not found.</p>
-    </Banner>);
+    return (
+      <Banner status="warning" title="Not found" my={10}>
```
Comment (shrirupa) [inline src/pages/api-keys/DetailsPage.js]: this line changed

File: src/pages/api-keys/DetailsPage.js [hunk 15]
```diff
@@ -31,22 +32,19 @@ export class ApiKeysDetailsPage extends Component {
...
     const { apiKey } = this.props;
 
-    return (<Banner
-      status='info'
-      title='This API Key is read-only'
-    >
-      <p>This API Key is only editable by the owner: {apiKey.username}.</p>
-    </Banner>);
+    return (
+      <Banner status="info" title="This API Key is read-only" my={10}>
```
Comment (shrirupa) [inline src/pages/api-keys/DetailsPage.js]: this line changed

File: src/components/verifyEmailBanner/VerifyEmailBanner.js [hunk 16]
```diff
@@ -28,7 +28,7 @@ export class VerifyEmailBanner extends Component {
     }
 
     return (
-      <Banner action={action} status="info" title="Verify your email address">
+      <Banner action={action} status="info" title="Verify your email address" my={10}>
```
Comment (shrirupa) [inline src/components/verifyEmailBanner/VerifyEmailBanner.js]: this line changed

File: src/components/apiErrorBanner/ApiErrorBanner.js [hunk 17]
```diff
@@ -38,17 +38,28 @@ export default class ApiErrorBanner extends Component {
     const showDetailsButton = errorDetails || payload.message;
 
     return (
-      <Banner status={status} title={title}>
+      <Banner status={status} title={title} my={10}>
```
Comment (shrirupa) [inline src/components/apiErrorBanner/ApiErrorBanner.js]: this line changed

File: src/components/apiKeyBanner/ApiKeyBanner.js [hunk 18]
```diff
@@ -10,40 +10,27 @@ export class ApiKeySuccessBanner extends Component {
...
     };
 
     return (
-      <Banner
-        action={action}
-        title={title}
-        status={status}
-        onDismiss={hideNewApiKey}
-      >
+      <Banner action={action} title={title} status={status} onDismiss={hideNewApiKey} my={10}>
```
Comment (shrirupa) [inline src/components/apiKeyBanner/ApiKeyBanner.js]: this line changed

File: src/components/matchbox/Banner.js [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+import { useHibana } from 'src/context/HibanaContext';
+import _ from 'lodash';
+
+export default function Banner(props) {
+  const [state] = useHibana();
+  const { isHibanaEnabled } = state;
+  const preHibanaProps = ['status', 'title', 'onDismiss', 'actions', 'action', 'children'];
+
+  if (isHibanaEnabled) {
+    return <HibanaBanner {...props} />;
```
Comment (PatrickSison) [inline src/components/matchbox/Banner.js]: SHould this be using the same omit props pattern as progressbar?

Comment (shrirupa) [reply src/components/matchbox/Banner.js]: I will go ahead and rebase this branch.

**Pull request merged**